wandb/sdk/launch/agent/agent.py

```diff
@@ -36,6 +38,83 @@
 _logger = logging.getLogger(__name__)
 
 
+def init_pool_processes(jobs: Dict[str, int], lock: Lock) -> None:
```

Comment:
does jobs have to be Dict[str, int]? Could it be a Dict[str, Dict[str,str]]? Could we store information about the submitted runs status in there?

Reply:
Great idea - the subthread that started a run can poll its status, and update the jobs dict to relay status to the main thread, which can use that to keep agent status updated